cairo-contracts/packages/apps/src/transfer/erc20_call.cairo

```diff
@@ -42,17 +41,14 @@ pub impl ERC20ContractImpl of ERC20ContractTrait {
         salt: felt252,
         name: ByteArray,
         symbol: ByteArray,
+        decimals: u8,
         amount: u256,
         recipient: ContractAddress,
         owner: ContractAddress,
     ) -> ERC20Contract {
         let mut call_data = array![];
 
-        call_data.append_serde(name);
-        call_data.append_serde(symbol);
-        call_data.append_serde(amount);
-        call_data.append_serde(recipient);
-        call_data.append_serde(owner);
+        (name, symbol, decimals, amount, recipient, owner).serialize(ref call_data);
```

Comment:
I refactored the code here like this, as I feel, semantically, this is correct.

Also, we don't need `SerializeAppend` anymore.